### PR TITLE
Use a single transform to process all wildcards

### LINKED ISSUE #209: Harmonising wildcard processing
Currently there are 2 transforms dealing with wildcard processing:
- `process_uc_wildcards` deals with wildcards in `uc_t` tables;
- `process_wildcards` deals with wildcards in `tfm` tables and changes the dataframe with attribute data based on their content.

Would it be practical to:
- separate the modification logic of the dataframe with attribute data based on the content of `tfm` tables out of `process_wildcards`;
- combine wildcard processing done in `process_uc_wildcards` and `process_wildcards` in a single transform?

I believe these changes would also allow addressing #153 and #154 easier.

### PR DESCRIPTION
**Pull request by olejandro**
Closes #209

### CODE REVIEW

**Comment by olejandro** [comment]
@SamRWest this one does not do any optmisations from #210. It only extends `process_uc_wildcards` to processing all the tables with wildcards and removes any wildcard processing from the original `process_wildcards`.

**Comment by olejandro** [comment]
@siddharth-krishna I've done some clean-up along the way. The regression on Ireland is due to it (i.e. not processing `tfm_upd` in `process_flexible_import_tables`; still need to find out what to do about it).

**Comment by olejandro** [comment]
@SamRWest would be very interesting to try this on AusTIMES. The overall processing time on TUSM went down to 5 minutes.

**Comment by SamRWest** [comment]
Looks great, thanks @olejandro!  I'll test it soon.

File: xl2times/__main__.py [hunk 1]
```diff
@@ -119,12 +119,11 @@ def convert_xl_to_times(
         transforms.include_tables_source,
         transforms.merge_tables,
         transforms.complete_processes,
-        transforms.apply_more_fixups,
         transforms.process_units,
-        transforms.process_years,
         transforms.complete_commodity_groups,
-        transforms.process_uc_wildcards,
         transforms.process_wildcards,
+        transforms.apply_transforms,
```
Comment (siddharth-krishna) [inline xl2times/__main__.py:125]: Should we call this `process_transform_tables` or `apply_transform_tables`? In this context where everything above and below is a transform, apply transforms is a bit confusing.

File: xl2times/transforms.py [hunk 2]
```diff
@@ -2425,21 +2311,16 @@ def eval_and_update(
             total=len(updates),
             desc=f"Processing wildcard for {datatypes.Tag.tfm_ins_txt.value}",
```
Comment (siddharth-krishna) [inline xl2times/transforms.py:2312]: Perhaps we change this log line as well?

File: xl2times/datatypes.py [hunk 3]
```diff
@@ -161,14 +160,38 @@ class TimesModel:
...
+    def data_years(self) -> Set[int]:
+        """
+        data_years are years for which there is data specified.
+        """
+        data_years = set()
+        for attributes in [self.attributes, self.uc_attributes]:
+            if not attributes.empty:
+                data_years.update(attributes["year"].astype(int).values)
+        # Remove interpolation rules before return
+        return {y for y in data_years if y >= 1000}
```
Comment (siddharth-krishna) [inline xl2times/datatypes.py:179]: These changes mean that `data_years` etc are recomupted every time they are called as `model.data_years`. Did we want that behaviour for something in particular? Performance wise the new version could be slower, especially if `attributes` is a large table.

**Review by siddharth-krishna — APPROVED** [review]
Thanks, Olex! It's nice to refactor and use Sam's blazing fast code everywhere -- nice to improve perf on TUSM as well. :)

Just some minor feedback:

File: xl2times/datatypes.py (hunk 3 above)
Comment (olejandro) [reply xl2times/datatypes.py:179]: Currently we don't use `model.data_years` for any processing. It is only needed to generate a corresponding set for GAMS. I expect calling it once after all the processing is done should suffice. I guess we could cache the value?

**Pull request merged**